### PR TITLE
[instruments] HP34401A adding remote control command.

### PR DESCRIPTION
**Pull request by Rybok**
* Checking stop bit configuration, should be two according to spec.
    * Added parameter: remote_local_state, it is set to "REM" during init
      by the driver.
      Should be set to "LOC" to return to local front display control.

### CODE REVIEW

File: pymeasure/instruments/hp/hp34401A.py [hunk 1]
```diff
@@ -300,6 +303,25 @@ def init_trigger(self):
...
     # System related commands
+    remote_local_state = Instrument.setting(
+        "SYST:%s",
+        """ A string property that controls the remote/local state of the
+        function generator. Valid values are: LOC<AL>, REM<OTE>, RWL<OCK>.
+        This setting can only be set. """,
+        validator=strict_discrete_set,
+        values=["LOC", "LOCAL", "REM", "REMOTE", "RWL", "RWLOCK"],
+        map_values=False,
+    )
```
Comment (bmoneke) [inline pymeasure/instruments/hp/hp34401A.py:314]: Is "RWL" relevant? If not, you could change it to a boolean "remote_control_enabled", which makes it a lot easier to understand: `remote_control_enabled = True`.

File: pymeasure/instruments/hp/hp34401A.py [hunk 2]
```diff
@@ -77,6 +78,8 @@ def __init__(self, adapter, name="HP 34401A", **kwargs):
             asrl={'baud_rate': 9600, 'data_bits': 7, 'parity': 2},
             **kwargs
         )
+        if type(adapter) == SerialAdapter:
+            self.init_serial()
```
Comment (bmoneke) [inline pymeasure/instruments/hp/hp34401A.py:82]: The normal procedure is, that a user uses the default VISAAdapter or they create their own adapter instance. In the latter case, the user has to provide the correct settings to the adapter instance (for example the `SerialAdapter`).
Therefore, we do not add any other adapter specific things to an instrument except for the VISAAdapter via the `super().__init__()` call.

If the read/write termination is wrong, you might add it to the super init into the `asrl` dictionary.

Comment (Rybok) [reply pymeasure/instruments/hp/hp34401A.py:82]: Will fix it this week.

File: pymeasure/instruments/hp/hp34401A.py (hunk 1 above)
Comment (Rybok) [reply pymeasure/instruments/hp/hp34401A.py:311]: Yes, for long measurements I find RWL very useful to provide protection against accidental reset of the remote control mode.

File: pymeasure/instruments/hp/hp34401A.py [hunk 3]
```diff
@@ -74,7 +74,7 @@ def __init__(self, adapter, name="HP 34401A", **kwargs):
         super().__init__(
             adapter,
             name,
-            asrl={'baud_rate': 9600, 'data_bits': 7, 'parity': 2},
+            asrl={'baud_rate': 9600, 'data_bits': 8, 'parity': 0},
```
Comment (Rybok) [inline pymeasure/instruments/hp/hp34401A.py:77]: According to [hp34401manual.pdf](https://engineering.purdue.edu/~aae520/hp34401manual.pdf) these are the default settings. Actually it should be 2 stop bits but using pyvisa one should use the enum provided.

**Comment by Rybok** [comment]
# Test code

```
from pymeasure.instruments.hp import HP34401A
import time

# Create an instance of the HP34401A instrument
dmm = HP34401A("asrl/dev/ttyUSB0")

time.sleep(0.5)

#Important, without the REM no measurements will be read.
dmm.remote_local_state = "REM"

time.sleep(0.5)
print(f"ID: {dmm.id}")



dmm.function_ = "DCV"

for i in range(3):
    time.sleep(0.3)
    print(f"{i} Voltage: {dmm.reading}")  # -> Single float reading

time.sleep(0.3)
dmm.remote_local_state = "LOC"





```

File: pymeasure/instruments/hp/hp34401A.py (hunk 1 above)
Comment (BenediktBurger) [reply pymeasure/instruments/hp/hp34401A.py:311]: Another idea as a **suggestion**: You have two boolean properties: `remote_control_enabled`, `remote_lock_enabled`.

Btw. you do not have to specify `map_value=True`.

File: pymeasure/instruments/hp/hp34401A.py (hunk 3 above)
Comment (BenediktBurger) [reply pymeasure/instruments/hp/hp34401A.py:77]: you could specify the stop bits with `stop_bits=20` (afaik 20 is the integer value of the stop bits 2 enum in pyvisa).

File: pymeasure/instruments/hp/hp34401A.py (hunk 1 above)
Comment (Rybok) [reply pymeasure/instruments/hp/hp34401A.py]: Ok, fixed to use boolean properties. But, wouldn't work without map_values set to true. But I guess you meant it for the first case.

File: pymeasure/instruments/hp/hp34401A.py (hunk 3 above)
Comment (Rybok) [reply pymeasure/instruments/hp/hp34401A.py]: Added the stop bits plus a link to where this magic "20" number comes from.

File: pymeasure/instruments/hp/hp34401A.py (hunk 1 above)
Comment (BenediktBurger) [reply pymeasure/instruments/hp/hp34401A.py]: Sorry, I meant, `map_values=False` is the default value, therefore you do not have to specify it in the (outdated) code above.

File: pymeasure/instruments/hp/hp34401A.py [hunk 4]
```diff
@@ -300,6 +302,22 @@ def init_trigger(self):
     )
 
     # System related commands
+    remote_control_enabled = Instrument.control(
+        "SYST: ", "SYST:%s",
+        """Control whether remote is enabled.""",
```
Comment (BenediktBurger) [inline pymeasure/instruments/hp/hp34401A.py:307]:
```suggestion
        """Control whether remote control is enabled.""",
```

**Review by BenediktBurger — APPROVED** [review]
One tiny change left.
As a contributor, you can add yourself to the authors.txt file (already in this PR).

Thanks for your contribution.

**Comment by Rybok** [comment]
> May I merge your code?

Yes, please do! :)

**Comment by BenediktBurger** [comment]
Thanks for your contribution and patience.

**Pull request merged**